orchestrator/lib/otel/tracing_middleware.go

```diff
@@ -26,7 +27,7 @@ func HandlerWithTracing(tracer trace.Tracer, operationName string, handler http.
 
 		span.SetAttributes(semconv.HTTPStatusCodeKey.Int(wrapped.statusCode))
 		if wrapped.statusCode >= 400 {
-			span.SetStatus(codes.Error, http.StatusText(wrapped.statusCode))
+			Error(span, errors.New(http.StatusText(wrapped.statusCode)))
```

Comment:
I'm not sure `RecordError()` is right here (as done by `Error`), since there's not really an error?